pkg/rc/memory_backed.go

```diff
@@ -0,0 +1,231 @@
+package rc
+
+import (
+	"strings"
+	"time"
+
+	"github.com/square/p2/pkg/kp"
+	"github.com/square/p2/pkg/labels"
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/util"
+)
+
+// I just copied this from kp.Store because it was a huge interface and I really only care about this one method!
+type kpStore interface {
+	SetPod(key string, manifest pods.Manifest) (time.Duration, error)
+}
+
+type memoryRc struct {
+	manifestSha string
+	logger      logging.Logger
+
+	kpStore        kpStore
+	nodeApplicator labels.Applicator
+	podApplicator  labels.Applicator
+
+	nodeSelector    labels.Selector
+	manifest        *pods.Manifest
+	replicasDesired int
+	disabled        bool
+}
+
+// memoryRc implements the ReplicationController interface.
+var _ ReplicationController = &memoryRc{}
+
+func NewMemoryBacked(
```

Comment:
highly likely that it will be beneficial to document the implications of this being in memory only (not persisted, etc.)